airbyte-cdk/python/pyproject.toml

```diff
@@ -1,44 +1,107 @@
+documentation = "https://docs.airbyte.io/"
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Developers",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+    "License :: OSI Approved :: MIT License",
+    "Programming Language :: Python :: 3.8",
 ]
+keywords = ["airbyte", "connector-development-kit", "cdk"]
 
-[tool.flake8]
-extend-exclude = [
-    "*/lib/*/site-packages",
-    ".venv",
-    "build",
-    "models",
-    ".eggs",
-    "airbyte-cdk/python/airbyte_cdk/models/__init__.py",
-    "airbyte-cdk/python/airbyte_cdk/sources/declarative/models/__init__.py",
-    ".tox",
-    "airbyte_api_client",
-    "**/generated/*",
-]
-max-complexity = 20
-max-line-length = 140
-
-extend-ignore = [
-    "E203",  # whitespace before ':' (conflicts with Black)
-    "E231",  # Bad trailing comma (conflicts with Black)
-    "E501",  # line too long (conflicts with Black)
-    "W503",  # line break before binary operator (conflicts with Black)
-]
\ No newline at end of file
+[tool.poetry.dependencies]
+python = "^3.8.1"
+airbyte-protocol-models = "0.5.1"
+backoff = "*"
+cachetools = "*"
+Deprecated = "~1.2"
+dpath = "~2.0.1"
+genson = "1.2.2"
+isodate = "~0.6.1"
+Jinja2 = "~3.1.2"
+jsonref = "~0.2"
+jsonschema = "~3.2.0"
+pendulum = "<3.0.0"
+pydantic = "^1.10.8"
+pyrate-limiter = "~3.1.0"
+python-dateutil = "*"
+PyYAML = "^6.0.1"
+requests = "*"
+requests_cache = "*"
+wcmatch = "8.4"
+
+[tool.poetry.dev-dependencies]
+avro = "~1.11.2"
+cohere = "4.21"
+fastavro = "~1.8.0"
+freezegun = "*"
+langchain = "0.0.271"
+markdown = "*"
+mypy = "1.6.0"
+openai = { version = "0.27.9", extras = ["embeddings"] }
+pandas = "2.0.3"
+pdf2image = "1.16.3"
+"pdfminer.six" = "20221105"
+poethepoet = "^0.24.2"
+pyarrow = "~15.0.0"
+pyproject-flake8 = "^6.1.0"
+pytesseract = "0.3.10"
+pytest = "6.2.5"
+pytest-cov = "*"
+pytest-httpserver = "*"
+pytest-mock = "*"
+requests-mock = "*"
+Sphinx = "~4.2"
+sphinx-rtd-theme = "~1.0"
+tiktoken = "0.4.0"
+unstructured = { version = "0.10.27", extras = ["docx", "pptx"] }
+"unstructured.pytesseract" = ">=0.3.12"
+
+[tool.poetry.extras]
+file-based = ["avro", "fastavro", "pyarrow", "unstructured", "pdf2image", "pdfminer.six", "unstructured.pytesseract", "pytesseract", "markdown"]
+sphinx-docs = ["Sphinx", "sphinx-rtd-theme"]
+vector-db-based = ["langchain", "openai", "cohere", "tiktoken"]
+
+[tool.mypy]
+warn_unused_configs = true
+warn_redundant_casts = true
+ignore_missing_imports = true
+strict_equality = true
+check_untyped_defs = true
+disallow_untyped_decorators = false
+disallow_any_generics = true
+disallow_untyped_calls = true
+disallow_incomplete_defs = true
+disallow_untyped_defs = true
+warn_return_any = true
+follow_imports = "silent"
+no_implicit_reexport = false
+#plugins = ["pydantic.mypy", "pendulum", "pytest-mypy-plugins"]
+
+[[tool.mypy.overrides]]
+module = "airbyte_cdk.models"
+ignore_errors = true
+
+[tool.poe.tasks]
```

Comment:
I'd love to have high level poe tasks exposed like:
* `poe lint` which would run flake
* `poe build` which would run the code gen + `poetry build` 
* `poe type_check` which would run mypy
* `poe test` which would run all the test suites of this project

I also found handy to have a `poe pre-push` command to use locally which would run all the tasks listed above.

Reply:
Good ideas! Updated to add some more, now we have 
```
CONFIGURED TASKS
  build-code-generator-image
  generate-component-manifest-files
  generate
  build-package
  build
  flake-check
  lint
  type-check
  unit-test-with-cov
  check
  pre-push
  ```